provider/datagen/src/bin/fingerprint-data.rs

```diff
@@ -0,0 +1,131 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use clap::{App, Arg};
+use icu_provider_blob::BlobDataProvider;
+use sha2::{Digest, Sha256};
+use simple_logger::SimpleLogger;
+use std::fs;
+use std::io::Write;
+use walkdir::WalkDir;
+
+fn main() -> eyre::Result<()> {
+    let matches = App::new("ICU4X Data fingerprinter")
+        .version("0.0.1")
+        .author("The ICU4X Project Developers")
+        .about("Generate a fingerprint file from a data folder containing hashes of the data")
+        .arg(
+            Arg::with_name("VERBOSE")
+                .short("v")
+                .long("verbose")
+                .help("Requests verbose output"),
+        )
+        .arg(
+            Arg::with_name("DATA")
+                .short("d")
+                .long("data")
+                .takes_value(true)
+                .required(true)
+                .help(
+                    "Path to folder containing data generated by icu4x-datagen --format=dir, 
+                       or file containing data generated by icu4x-datagen --format=blob.",
+                ),
+        )
+        .arg(
+            Arg::with_name("FORMAT")
+                .long("format")
+                .takes_value(true)
+                .possible_value("dir")
+                .possible_value("blob")
+                .help("Output to a directory on the filesystem or a single blob.")
```

Comment:
Nit: update the doc. You can even remove the arg, we can detect whether a path is a dir or a file.